components/Reference/DynamicFee.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState, useContext } from 'react'
+
+import debounce from 'lodash.debounce'
+import { IOpcode } from 'types'
+
+import { EthereumContext } from 'context/ethereumContext'
+
+import { calculateDynamicFee } from 'util/gas'
+
+import { Input, Radio, Icon } from 'components/ui'
+import { H2 } from 'components/ui/Doc'
+
+const debounceTimeout = 100 // ms
+
+type Props = {
+  opcode: IOpcode
+}
+
+type InputValue = {
+  [name: string]: string
+}
+
+const DynamicFee = ({ opcode }: Props) => {
+  const { dynamicFee } = opcode
+
+  const { common } = useContext(EthereumContext)
+  const [inputs, setInputs] = useState<InputValue>({})
+  const [result, setResult] = useState('0')
+
+  // Initialize inputs with default keys & values
+  useEffect(() => {
+    const inputValues: InputValue = {}
+    Object.keys(dynamicFee?.inputs || []).map((key: string) => {
+      if (dynamicFee?.inputs[key].type === 'number') {
+        inputValues[key] = '0'
+      } else if (dynamicFee?.inputs[key].type === 'boolean') {
+        inputValues[key] = '1' // true default
```

Comment:
by default, we set booleans to 1 - True - Yes

Reply:
I think false as default would make more sense.

Reply:
> It's a good start, though I am missing the extra documentation for the dynamic gas cost portion (that would be also potentially fork dependent), where would it go? I am also missing the formula displayed, did you think about that?
> 
> Lastly, I think all the inputs take a lot of space, is there a way to condense that? Like for the boolean and the warm cold, I would prefer not to ask a question, just two radio buttons warm and cold, and for the boolean maybe just a button with "empty account" as label. The documentation will explain all that needs to be explained. Would love also that any input that is directly part of the formula be displayed directly in the formula, not taking an additional line. And that way the result gas cost would be the last term of the formula, right now I find it counter intuitive that it be in the title.
> 
> Is that feasible?

Re "missing dynamic gas cost" - we have a handler to the `common` object, so everything can be picked up from there.

Re "formulas" - Some formulas are not single-line that you can easily display, isn't it? So, I have a hard time imagining how it would look in the UI. As you remember, we first tried to do inline-editing of the formulas, but it didn't work for all opcodes.

Reply:
> missing dynamic gas cost
I have to say I do not understand your answer. I am asking where do I write the documentation that I want to add for each opcode with dynamic gas cost. Markdown description. Not something we get from the code.

> formulas
Yes for the formulas not all can, but can we make the ones that can inlined with their parameters? Or is it too much effort? (Remember I am asking here not as a technical person, if it's too hard to do for some technical reason you know best and you can say so)

> condensed representation
An idea could be to put two variables per line also, for example, the numbers do not need all that horizontal space.

Reply:
> missing dynamic gas cost

Aga, gotcha. So, if it's a simple line of text, we can add it in the opcode.json as description metadata and show it in the UI near each input. If it's more detailed and needs markdown, then we add it to the opcode markdown files, right in the top as usual, with some heading.

> formulas

Of course, technically it's feasible, but the question is if it makes sense UI/UX-wise, worth the implementation effort, and code maintainability. If we were to go that route, then each formula will have to be displayed differently in the UI, some will be a single line of editable text, another multi-line inputs with radios, etc. This makes the UI inconsistent and code more brittle, as now you have to define and build it differently. With the current approach, it's all driven by a single metadata file and the UI is auto-generated for any kind of opcode (regardless of how many formula terms there are), in a consistent manner. It's simple and serves the purpose. So, IMHO, it's not worth the effort to show a few opcodes as formulas with inline-editing capability.

Reply:
> missing dynamic gas cost

It's unlikely to be a simple line of text, and it does not fit in the opcodes markdown files, as we need it to be versioned. Though I guess if you can add a filter in the markdown so that we can write all descriptions but you only display the one from the current fork, that sounds nice to me.

> formulas

That's fair, let's keep it that way then. Let's just adapt how we display the inputs. For example, instead of a line for name and then a line for the value, can we try to put them both on the same line (just thinking that if that takes too much vertical space, it's kind of hiding what's below). I am also wondering what would be the better placement. I think maybe it would be better that the dynamic gas cost be the last of the extra info, that way it's less problematic when it takes so much vertical space, and the description of the opcode should probably be first anyway.

To display the formulas, would be awesome if I could write them in the markdown too. If it's versioned by fork, I guess I can hardcode the constant costs and we don't need to read them dynamically from ethereum js. Though I guess it would still be convenient if nothing changed but the constant values, would not have to copy paste everything but the constants. And would still need a placeholder to write the result of the formula (the title is not the best place).

Would be nice in that case also to have a way to have a "default", or maybe even better, the filter would take the oldest hardfork it can find. That way, if it did not change, I don't have to copy paste for each version.

I am asking a lot, sorry :-)
Is any of those realistic?

Reply:
So after some discussion and thoughts with @peroket, here is what we will do to improve doc experience around dynamic gas:

1. Move dynamic gas calculation form to the side (floating), and on narrower screens place at the bottom of the doc.
2. Add support for dynamic variables in the opcode markdown documents, starting with EthereumJS Common params. We will namespace them with `eths` and wrap in curly braces, for instace `expByte` from the library would be `{ethjs|expByte}` in the doc.
3. We will start with the same documents, but if it appears that there is a substantial difference in gas calculation formulas & documentation between the forks, we will add doc overrides, for example: `0a/london.mdx`